examples/molasses_1d.rs

```diff
@@ -20,13 +20,13 @@ fn main() {
 
     // Add some output to the simulation
     builder = builder.with(
-        file::new::<Velocity, Text>("vel.txt".to_string(), 10),
+        file::new::<Velocity, Text, Atom>("vel.txt".to_string(), 10),
```

Comment:
Is it neccessary to have all of these API breaking changes? Is it possible to instead define `file::new` so that the default implementation assumes the type is `Atom`, e.g. using Default Generic Type Parameters? That way the new functionality is added but without breaking it for existing users.

Reply:
I wanted to do that originally but couldn't find a way to express defaults for types in Rust. (The compiler always told me that you cant do that in rust)

Reply:
Can you provide two function implementations, eg `new::<C,O,F>` and `new::<C,O>` and have one implementation return the other with `Atom` as third param?

Reply:
It appears that rust does not support this kind of function overloading in general. Also, I found no way that you can make the number of arguments passed in the <> brackets variable in any way...



Possible workarounds would be: To be in a different namespace or to give the function a different name.

Reply:
perhaps you could give the new functionality a different name, eg `new_with_filter`

Reply:
fine with me, I'll do that.

Reply:
done